src/tools/rest-backend/service_config.cpp

```diff
@@ -36,12 +39,73 @@ cppcms::json::value ConfigEngine::loadApplicationConfiguration (const std::strin
 	kdb.get (ks, conf_root);
 	ks = ks.cut (kdb::Key (conf_root, KEY_END));
 
-	for (auto elem : ks)
+	cppcms::json::value result = this->transformKeysetToJsonValue (ks, conf_root);
+
+	return result;
+}
+
+/**
+ * @brief can be used to transform a kdb::KeySet into cppcms::json::value
+ * 
+ * will iterate through the keyset and use a helper method to add the key
+ * values to the json::value
+ * 
+ * @param ks the keyset that needs to be transformed
+ * @return a cppcms::json::value containing the configuration from the keyset
+ */
+cppcms::json::value ConfigEngine::transformKeysetToJsonValue (const kdb::KeySet & ks, const std::string & conf_root) const
+{
+	cppcms::json::value result;
+	kdb::KeySet ksDup = ks.dup ();
+	kdb::KeySet ksToVisit = ksDup.cut (kdb::Key ("spec/", KEY_END));
+
+	// all spec keys
+	for (auto elem : ksToVisit)
 	{
-		std::string key = elem.getName ().substr (elem.getName ().find (conf_root) + conf_root.length () + 1);
-		std::replace (key.begin (), key.end (), '/', '.');
+		kdb::Key elemCascading = elem.dup ();
+		kdb::tools::helper::removeNamespace (elemCascading);
+		kdb::Key key = ksDup.lookup (elemCascading.getName ());
+		if (!key)
+		{
+			key = elem;
+		}
 
-		this->setValue (result, key, elem);
+		std::string keyName = key.getName ().substr (key.getName ().find (conf_root) + conf_root.length () + 1);
+		std::replace (keyName.begin (), keyName.end (), '/', '.');
+
+		// set default value in case elem has no value
+		if ((key.getBinarySize () == 0 || key.getString () == "") && key.hasMeta ("default"))
```

Comment:
It is not implemented within `kdbGet` as discussed in #1152.

But `ksLookup` of course implements all these specifications as described in `doc/help/elektra-cascading.md` or in `ksLookup`'s API docu. (This might add new default keys to the keyset, which is not described in the API docu because we hopefully can get rid of this behavior.).

I updated the docu to better describe this.